### PR TITLE
feat: Better text formatting

### PR DESCRIPTION
**Pull request by callinmullaney**
### Purpose:
- Improve default configuration for all text formats to be more purposeful. 
- Replace the `Upload Image` widget with embedded media Image with an entity browser
- Configure `Restricted Html` filter for specific use case that allow for limited formatting. Bold, Italics, Link for simple components like cards and Page intro text.
- Add another image based entity browser that is configured with an iframe instead of a modal. This is specifically for embedded media support. 

### Testing:
- [x] Install Sous and verify the config changes in this PR import without errors.
- [x] Go to `/node/add/page` and add a Text component. Verify the following:
    - [x] The Image widget is replaced with `Embedded Image Browser`. This is the `E` icon - see note below
    - [x] Clicking this widget triggers an entity browser specific to images that is identical to the one we use for field wigets
    - [x] Embedding an image media entity renders immediately in CkEditor
- [x] Switch to Full_Html and verify there is additional WYSIWYG formatting widgets included the `Embedded Image Browser`
- [x] Switch to Restricted and verify it's a limited ckeditor with Bold, Italics, Link, and Source option. 

### Notes:
- The `E` icon for the embedded Image Browser can be customized here: `/admin/config/content/embed/button/manage/embedded_image_browser` However, the file is saved to the files folder that generally isn't migrated or deployed. It can be customized on a per project basis for now.
- **Full HTML Filter**: An unrestricted filter that only corrects broken html and removes empty `p` tags when rendered.
- **Basic HTML Filter**: General use filter that offers a broad range of formatting options. Ideal for most instances of formatted text.
- **Restricted HTML Filter**: Limited access filter best used for specific scenarios where some formatting is appropriate. 
- **Plain Text Filter**: Rendered as a raw string.

### CODE REVIEW

**Comment by americkson** [comment]
This "working" as expected from the testing steps but there seems to be an unnecessary step in selecting an image. Upon clicking the "e" Icon, I get this:

Screen Shot 2021-07-02 at 12 34 17 PM

Which I have the click the 'Select Images' link. Then I get the actual image browser:

Screen Shot 2021-07-02 at 12 34 27 PM

I wonder if we should have a different  image browser configuration for this use case?

The only other thing is I wonder if we can replace that default 'E' icon to at least be an "I" OR we add a hook install step to get an icon that represents embedding an image and have it there by default? We could separate that out as a different issue too.

It will be good to get the entity browser fixes merged in (different issue) then explore what it might take to get that extra step out of the process.

**Review by americkson — CHANGES_REQUESTED** [review]
See post above. Let's discuss at some point soon.

File: config/install/entity_browser.browser.image_browser_embedded.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - media
+    - views
+name: image_browser_embedded
+label: "Image Browser (Embedded)"
+display: iframe
+display_configuration:
+  width: 100%
+  height: "650"
+  link_text: "Select Image(s)"
+  auto_open: true
```
Comment (callinmullaney) [inline config/install/entity_browser.browser.image_browser_embedded.yml:18]: @americkson This auto-open setting should be skipping that extra step of `select image(s)`. It's doing it for me on my end. Can you verify this setting is active for the `Image Browser (Embedded)` entity browser?

**Comment by callinmullaney** [comment]
@americkson Currently, the `E` icon cannot be changed and remain customized across environments. The problem is the customized icon is stored as a managed file and not something exportable. This is a known issue : https://www.drupal.org/project/embed/issues/2824110#comment-13055494 

Should I include this patch in this PR and upload a custom icon or should we wait until this is fully tested and merged?

**Review by ccjjmartin — APPROVED** [review]
@callinmullaney We can revert my change out if you want but when I reviewed this I noticed that no images I embedded were actually showing.  When I dug into it I found that there is a separate text filter that you have to enable in order for the embedded images to be rendered.  So I enabled it and the images are now showing for me (rather than having the html `<drupal-entity ...` in the rendered output of the page.  It could be due to the fact that I recently updated a lot of modules in other PRs so probably a new "feature".  Anyway, after that one change this PR looks good to me :+1:

File: config/install/entity_browser.browser.image_browser_embedded.yml (hunk 1 above)
Comment (ccjjmartin) [reply config/install/entity_browser.browser.image_browser_embedded.yml:18]: It is skipping that step for me when I tried it, I would call this one resolved.

**Comment by ccjjmartin** [comment]
@americkson I think Callin's approach here of letting each site replace the E icon is acceptable and would call this one ready to merge since it has a lot of good stuff.  If we want the E icon changed let's do that in a follow up PR?

**Review by americkson — APPROVED** [review]

**Pull request merged**